python/cudf/utils/cudautils.py

```diff
@@ -184,6 +184,12 @@ def fill_value(arr, value):
     gpu_fill_value.forall(arr.size)(arr, value)
 
 
+def full(size, value, dtype):
+    out = rmm.device_array(size, dtype=dtype)
+    fill_value(out, value)
+    return out
```

Comment:
We need to check either here or in the `fill_value` function if `out` is size 0 before calling the kernel.

Reply:
done